### DISCUSSION TITLE
How to add a rule defined in JSON into Knowledge Library

### DISCUSSION THREAD

**Discussion by kalyangouthammekala** [hyperjumptech/grule-rule-engine#238] [Q&A] [answered]
Hi, I wish to check an example to load the rule in Json format as a string. If possible please share a whole example like we have for AgeCheckSample_test. Thanks,

When i try i get the following error.
grl error on 2:1 mismatched input '{' expecting {<EOF>, RULE}"

sample json rule:
const ruleInJson = `
	{
  "name": "RuleToCheck",
  "desc": "rule in json",
  "salience": 10,
  "when": "Pogo.Result == false",
  "then": [
    "Pogo.Result = true"
  ]
}

**Comment by newm4n** [comment]
I assume you have not read this ? https://github.com/hyperjumptech/grule-rule-engine/blob/master/docs/GRL_JSON_en.md

**Reply by kalyangouthammekala** [reply]
Hi I tried it by adding the json as a string into a constant just like we do with GRL and received the above error.
grl error on 2:1 mismatched input '{' expecting {, RULE}"

It works for GRL but not for JSON.
I wish to know if i am missing something.

**Comment by kalyangouthammekala** [comment, marked as answer]
Got it to work..using the lines below
ruleset, err := pkg.ParseJSONRuleset([]byte(ruleInJson))
.
.
err = ruleBuilder.BuildRuleFromResource("Test", "0.1.1", pkg.NewBytesResource([]byte(ruleset)))
Thanks.
